src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/ASPNetBenchmarks/AspNetBenchmarksCommand.cs

```diff
@@ -42,11 +42,167 @@ public override int Execute([NotNull] CommandContext context, [NotNull] AspNetBe
 
             ConfigurationChecker.VerifyFile(settings.ConfigurationPath, nameof(AspNetBenchmarksCommand));
             ASPNetBenchmarksConfiguration configuration = ASPNetBenchmarksConfigurationParser.Parse(settings.ConfigurationPath);
+
+            // Before running the ASP.NET benchmarks, execute a few checks:
+            // Check 1. If you are connected to corpnet. These tests only run if you are connected to corp-net and therefore, preemptively, check this.
+            // Check 2. The host machines reboot between 12:00 AM - 12:08 AM PST. Check if we are running during that time, and if so, delay running the infrastructure.
+
+            // Check 1.
+            if (!Directory.Exists(@"\\clrmain\tools\"))
+            {
+                // Not connected to corpnet if that folder is unavailable.
+                AnsiConsole.MarkupLine($"[red bold] Not connected to corpnet. Ensure you are connected before proceeding with the ASP.NET Benchmarks [/]");
+                return -1;
+            }
+
+            // Check 2.
+            SleepUntilHostsHaveRestarted();
+
             RunASPNetBenchmarks(configuration);
             AnsiConsole.MarkupLine($"[bold green] Report generated at: {configuration.Output.Path} [/]");
             return 0;
         }
 
+        private static void SleepUntilHostsHaveRestarted()
+        {
+            DateTime now = DateTime.UtcNow;
+
+            // Get the Pacific Standard Time zone (considering daylight saving).
+            TimeZoneInfo pstZone = TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time");
+            DateTime pstNow = TimeZoneInfo.ConvertTimeFromUtc(now, pstZone);
+
+            // Check if the current time is between 12:00 AM and 12:08 AM.
+            DateTime start = pstNow.Date; // 12:00 AM today.
+            DateTime end = start.AddMinutes(8); // 12:08 AM today.
+
+            if (pstNow >= start && pstNow < end)
+            {
+                TimeSpan timeUntilEnd = end - pstNow;
+                int secondsLeft = (int)timeUntilEnd.TotalSeconds + 60;
+
+                // If we are between 12:00 AM and 12:08 AM PST, sleep for 
```

Comment:
```suggestion
            // Check if the current time is between 12:00 AM and 12:09 AM.
            DateTime start = pstNow.Date; // 12:00 AM today.
            DateTime end = start.AddMinutes(9); // 12:09 AM today.

            if (pstNow >= start && pstNow < end)
            {
                TimeSpan timeUntilEnd = end - pstNow;
                int secondsLeft = (int)timeUntilEnd.TotalSeconds;

                // If we are between 12:00 AM and 12:09 AM PST, sleep for 
```

Reply:
(just being consistent about 12:07:59 and 12:08:01)

Reply:
also a question - should this sleep at 11:59PM as well, or will the retry logic handle that?  Starting a long run at something like 11:30PM is useful :)

Reply:
Retry waits 2 minutes, so I think attempt #2 could hit the 12:00-12:08 window.

Reply:
As a heads up, added this check at the beginning of each benchmark run - this will take care of the issue with hitting the restart period without retrying twice.